src/compiler/rust.rs

```diff
@@ -1184,6 +1427,239 @@ impl pkg::ToolchainPackager for RustToolchainPackager {
     }
 }
 
+#[cfg(feature = "dist-client")]
+struct RustOutputsRewriter {
+    dep_info: Option<PathBuf>,
+}
+
+#[cfg(feature = "dist-client")]
+impl OutputsRewriter for RustOutputsRewriter {
+    fn handle_outputs(self: Box<Self>, path_transformer: &dist::PathTransformer, output_paths: Vec<(String, PathBuf)>)
+                      -> Result<()> {
+        use std::io::{Seek, Write};
+
+        // Outputs in dep files (the files at the beginning of lines) are untransformed - remap-path-prefix is documented
+        // to only apply to 'inputs'.
+        trace!("Pondering on rewriting dep file {:?}", self.dep_info);
+        if let Some(dep_info) = self.dep_info {
+            for (_dep_info_dist_path, dep_info_local_path) in output_paths {
+                trace!("Comparing with {}", dep_info_local_path.display());
+                if dep_info == dep_info_local_path {
+                    error!("Replacing using the transformer {:?}", path_transformer);
+                    // Found the dep info file
+                    let mut f = fs::OpenOptions::new()
+                        .read(true)
+                        .write(true)
+                        .open(dep_info)?;
+                    let mut deps = String::new();
+                    f.read_to_string(&mut deps)?;
+                    for (local_path, dist_path) in get_path_mappings(path_transformer) {
+                        let re_str = format!("(?m)^{}", regex::escape(&dist_path));
+                        error!("RE replacing {} with {} in {}", re_str, local_path.to_str().unwrap(), deps);
+                        let re = regex::Regex::new(&re_str).expect("Invalid regex");
+                        deps = re.replace_all(&deps, local_path.to_str().unwrap()).into_owned();
+                    }
+                    f.seek(io::SeekFrom::Start(0))?;
+                    f.write_all(deps.as_bytes())?;
+                    f.set_len(deps.len() as u64)?;
+                    return Ok(())
+                }
+            }
+            // We expected there to be dep info, but none of the outputs matched
+            bail!("No outputs matched dep info file {}", dep_info.display());
+        }
+        Ok(())
+    }
+}
+
+
+#[cfg(feature = "dist-client")]
+#[derive(Debug)]
+struct RlibDepsDetail {
+    deps: Vec<String>,
+    mtime: time::SystemTime,
+}
+
+#[cfg(feature = "dist-client")]
+struct DepsSize;
+#[cfg(feature = "dist-client")]
+impl lru_cache::Meter<PathBuf, RlibDepsDetail> for DepsSize {
+    type Measure = usize;
+    fn measure<Q: ?Sized>(&self, _k: &Q, v: &RlibDepsDetail) -> usize where PathBuf: Borrow<Q>
+    {
+        use std::mem;
+
+        // TODO: unfortunately there is exactly nothing you can do with the k given the
+        // current trait bounds. Just use some kind of sane value;
+        //let k_size = mem::size_of::<PathBuf>() + k.capacity();
```

Comment:
I think this was mostly to account for the memory usage case where you would want to measure both the keys and values. If you only care about the values this is fine.